include/rmm/detail/error.hpp

```diff
@@ -55,11 +55,11 @@
   GET_RMM_EXPECTS_MACRO(__VA_ARGS__, RMM_EXPECTS_3, RMM_EXPECTS_2) \
   (__VA_ARGS__)
 #define GET_RMM_EXPECTS_MACRO(_1, _2, _3, NAME, ...) NAME
-#define RMM_EXPECTS_3(_condition, _reason, _exception_type)                       \
-  (!!(_condition)) ? static_cast<void>(0)                                         \
-                   : throw _exception_type /*NOLINT(bugprone-macro-parentheses)*/ \
-  {                                                                               \
-    "RMM failure at: " __FILE__ ":" RMM_STRINGIFY(__LINE__) ": " _reason          \
+#define RMM_EXPECTS_3(_condition, _reason, _exception_type)                             \
+  (!!(_condition)) ? static_cast<void>(0)                                               \
+                   : throw _exception_type /*NOLINT(bugprone-macro-parentheses)*/       \
+  {                                                                                     \
+    std::string("RMM failure at: " __FILE__ ":" RMM_STRINGIFY(__LINE__) ": ") + _reason \
```

Comment:
Builds are failing like:
```cpp
$SRC_DIR/benchmarks/multi_stream_allocations/multi_stream_allocations_bench.cu(143): error: expected an expression
    throw rmm::logic_error{std::string{"RMM failure at:" "$SRC_DIR/benchmarks/multi_stream_allocations/multi_stream_allocations_bench.cu" ":" "143" ": "} + };;
```

I think the use of a `std::string` here imposes some additional requirements on the syntax. I'll push a fix if I can, I think it may just need `+` or fewer quotes.

Reply:
Yes, the problem is that using `std::string` here no longer works if we have zero arguments passed, like `RMM_FAIL();`. I will push a fix that supports 0-argument usage.

Reply:
I'm changing my mind. I don't want to support the zero-argument flavor of this. It is not documented as supported right now. I will fix `multi_stream_allocations_bench` to have a `_what` message instead.

Reply:
Fixed in 8fe497e.